### PR TITLE
[flutter_tools] refactor stringsArg

### PR DESCRIPTION
**Pull request by amanv8060**
Update stringsArg function.

part of #101595

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by amanv8060** [comment]
cc/ @Jasguerrero

File: packages/flutter_tools/lib/src/runner/flutter_command.dart [hunk 1]
```diff
@@ -1544,7 +1544,12 @@ abstract class FlutterCommand extends Command<void> {
   int? intArg(String name) => argResults?[name] as int?;
 
   /// Gets the parsed command-line option named [name] as `List<String>`.
-  List<String> stringsArg(String name) => argResults?[name] as List<String>? ?? <String>[];
+  List<String> stringsArg(String name) {
```
Comment (Jasguerrero) [inline packages/flutter_tools/lib/src/runner/flutter_command.dart:1547]: After this issue https://github.com/flutter/flutter/issues/105263#issuecomment-1145435487 we are going to change this to bubble up the error when the flag is missing so the behavior we are looking here will be something like `arResults?[name]! as List<String>? ?? <String>[]`

Comment (amanv8060) [reply packages/flutter_tools/lib/src/runner/flutter_command.dart:1544]: Sure, Changed to `argResults![name]! as List<String>? ?? <String>[];`

Comment (amanv8060) [reply packages/flutter_tools/lib/src/runner/flutter_command.dart:1544]: Did not change it back to arrow operator, based upon [styleguide section on arrow operators](https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#consider-using--for-short-functions-and-methods).

Comment (christopherfujino) [reply packages/flutter_tools/lib/src/runner/flutter_command.dart:1544]: LGTM

**Review by christopherfujino — APPROVED** [review]

**Review by Jasguerrero — APPROVED** [review]
LGTM

**Comment by amanv8060** [comment]
@Jasguerrero can you please add the autosubmit label over here ?

**Comment by christopherfujino** [comment]
@CaseyHillers what should we do here? It's doing the thing where the frob dashboard doesn't hvae the PR, but I get a failure if I try to schedule it.

**Comment by CaseyHillers** [comment]
@amanv8060 this PR is 800 commits behind tip of tree. Can you rebase to the latest to fix the stuck Google Testing check?

**Comment by amanv8060** [comment]
@christopherfujino  tests have passed.

**Pull request merged**